system/lib/wasmfs/syscalls.cpp

```diff
@@ -513,4 +520,85 @@ __wasi_errno_t __wasi_fd_fdstat_get(__wasi_fd_t fd, __wasi_fdstat_t* stat) {
   }
   return __WASI_ERRNO_SUCCESS;
 }
+
+// This enum specifies whether rmdir or unlink is being performed.
+enum class UnlinkMode { Rmdir, Unlink };
+
+static long __unlink(char* path, UnlinkMode unlinkMode) {
+  auto pathParts = splitPath(path);
+
+  // Root (/) directory cannot be removed.
+  if (pathParts.size() == 1 && pathParts[0] == "/") {
+    return -EBUSY;
+  }
```

Comment:
It would be better to check this by pointer comparison with the root dir, since there might be other paths that resolve to the root, e.g. using symlinks or `..`.

Reply:
This can be part of general refactoring for path parsing. Made a todo for this.